### DISCUSSION TITLE
Non generic ArchetypeQuery missing ForEach ForEachEntity

### DISCUSSION THREAD

**Discussion by dvdking** [friflo/Friflo.Engine.ECS#3] [Q&A] [answered]
I am trying to do such query without components needed.
```
      _store.Query()
             .AllTags(Tags.Get<PlayerInput>())
             .ForEachEntity((ref PlayerInput input, Entity entity) => { entity.DeleteEntity(); });
```
I am using 2.2.0 lib from nuget.
Is there any reason there is no ForEach ForEachEntity methods for non generic query?

**Comment by friflo** [comment, marked as answer]
hi @dvdking,

good question - I felt into the same trap once.
In case you don't iterate components you can simply use `query.Entities`.

```cs
var query = store.Query().AllTags(Tags.Get<PlayerInput>());
foreach (var entity in query.Entities) {
    entity.DeleteEntity(); // should be deleted by a command buffer - but different topic.
}
```
Using this approach is / was by intention.
`foreach()` is simpler in usage & understanding and significant faster that using delegates in `ForEach()` or `ForEachEntity()`.

**Comment by friflo** [comment]
As follow up of this feedback:

Added a `foreach(...)` to the `EntityQueries` example at
https://friflo.gitbook.io/friflo.engine.ecs/examples/general#query-entities
